### PR TITLE
Add support for pulsing lights

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

adds PulsingLightComponent, which makes a light's energy and radius grow and shrink in a regular interval based on a sine graph.
Just a nice tool for making some moody setpieces or something.

**Media**

image
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

no cl no fun

### CODE REVIEW

File: Content.Client/Light/PulsingLightSystem.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        var brightness = bA * MathF.Sin(b * val) + bD;
+        var radius = rA * MathF.Sin(b * val) + rD;
+
+        light.Energy = brightness;
+        _pointLight.SetRadius(uid, radius, light);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
```
Comment (ElectroJr) [inline Content.Client/Light/PulsingLightSystem.cs:47]: FrameUpdate

**Comment by ElectroJr** [comment]
What is the plan with the old animation code? Are we just removing it. Cause currently I'm pretty sure this just does the same thing as `PulseBehaviour`, so if we are keeping animations then this system should just use that instead.

**Comment by metalgearsloth** [comment]
The old lighting stuff is a big PITA to use and couples expendablelights with the animations too but that would need merging into this.

**Comment by EmoGarbage404** [comment]
converted the old ones to the new system

**Comment by EmoGarbage404** [comment]
gonna close this for now because the original system is more thorough and i don't feel like rewriting all of it to be components.

**Pull request closed without merging**